src/ChickenAPI.Packets/GenericBasePacketDeserializer.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace ChickenAPI.Packets
+{
+    public abstract class GenericBasePacketDeserializer<TPacket> : IPacketDeserializer where TPacket : IPacket
+    {
+        protected abstract string Header { get; }
+
+        public IPacket Deserialize(string buffer)
+        {
+            if (!buffer.StartsWith(Header))
```

Comment:
you may get error with things like #packet

Reply:
What's your suggestion for that ?
Putting a property "IsReturnablePacket" inside the deserializer to also check with `$"#{Header}"` ?

Reply:
yes maybe a good solution